报错：pydantic_core._pydantic_core.ValidationError: 7 validation errors for Song

使用指令`/m b50`报错，虽然看意思是请求次数太多，但是我试了2个小时也没成功输出过一次。
其次建议对这个报错捕捉异常输出，提示错误原因）

```
06-10 17:04:59 [DEBUG] nonebot | Plugin-Alconna | Alconna::mai 对 "/m b50" 的解析结果是 (matched=True, header_match=(origin='/m' result='/mai' matched=True groups={}), options={'best50': (value=Ellipsis args={})})
06-10 17:04:59 [INFO] nonebot | Event will be handled by AlconnaMatcher(type='', module=nb2.plugins.nonebot_plugin_lxns_maimai, lineno=57)
06-10 17:04:59 [DEBUG] nonebot | Running AlconnaMatcher(type='', module=nb2.plugins.nonebot_plugin_lxns_maimai, lineno=57)
06-10 17:04:59 [DEBUG] nonebot | Running handler Dependent(call=_, parameterless=(Depends(Dependent(call=_arparma_check), use_cache=False),))
06-10 17:04:59 [DEBUG] nonebot | Handler Dependent(call=_, parameterless=(Depends(Dependent(call=_arparma_check), use_cache=False),)) skipped
06-10 17:04:59 [DEBUG] nonebot | Running handler Dependent(call=_, parameterless=(Depends(Dependent(call=_arparma_check), use_cache=False),))
06-10 17:05:08 [INFO] nonebot | AlconnaMatcher(type='', module=nb2.plugins.nonebot_plugin_lxns_maimai, lineno=57) running complete
06-10 17:05:08 [ERROR] nonebot | Running AlconnaMatcher(type='', module=nb2.plugins.nonebot_plugin_lxns_maimai, lineno=57) failed.
Traceback (most recent call last):
  File "<string>", line 17, in <module>
  File "D:\截图\其他文件\nb2\.venv\Lib\site-packages\nonebot\__init__.py", line 337, in run
    get_driver().run(*args, **kwargs)
  File "D:\截图\其他文件\nb2\.venv\Lib\site-packages\nonebot\drivers\fastapi.py", line 187, in run
    uvicorn.run(
  File "D:\截图\其他文件\nb2\.venv\Lib\site-packages\uvicorn\main.py", line 580, in run
    server.run()
  File "D:\截图\其他文件\nb2\.venv\Lib\site-packages\uvicorn\server.py", line 66, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "D:\Program Files (x86)\python\Lib\asyncio\runners.py", line 195, in run
    return runner.run(main)
  File "D:\Program Files (x86)\python\Lib\asyncio\runners.py", line 118, in run
    return self._loop.run_until_complete(task)
  File "D:\Program Files (x86)\python\Lib\asyncio\base_events.py", line 678, in run_until_complete
    self.run_forever()
  File "D:\Program Files (x86)\python\Lib\asyncio\windows_events.py", line 322, in run_forever
    super().run_forever()
  File "D:\Program Files (x86)\python\Lib\asyncio\base_events.py", line 645, in run_forever
    self._run_once()
  File "D:\Program Files (x86)\python\Lib\asyncio\base_events.py", line 1999, in _run_once
    handle._run()
  File "D:\Program Files (x86)\python\Lib\asyncio\events.py", line 88, in _run
    self._context.run(self._callback, *self._args)
  File "D:\截图\其他文件\nb2\.venv\Lib\site-packages\nonebot\utils.py", line 254, in run_coro_with_shield
    return await coro
  File "D:\截图\其他文件\nb2\.venv\Lib\site-packages\nonebot\message.py", line 506, in check_and_run_matcher
    await _run_matcher(
> File "D:\截图\其他文件\nb2\.venv\Lib\site-packages\nonebot\message.py", line 458, in _run_matcher
    await matcher.run(bot, event, state, stack, dependency_cache)
  File "D:\截图\其他文件\nb2\.venv\Lib\site-packages\nonebot\internal\matcher\matcher.py", line 926, in run
    await self.simple_run(bot, event, state, stack, dependency_cache)
  File "D:\截图\其他文件\nb2\.venv\Lib\site-packages\nonebot\internal\matcher\matcher.py", line 863, in simple_run
    await handler(
  File "D:\截图\其他文件\nb2\.venv\Lib\site-packages\nonebot\dependencies\__init__.py", line 113, in __call__
    return await cast(Callable[..., Awaitable[R]], self.call)(**values)
  File "D:\截图\其他文件\nb2\nb2\plugins\nonebot_plugin_lxns_maimai\__init__.py", line 124, in _
    await UniMessage.image(raw=await render_b50(props)).finish(at_sender=True)
  File "D:\截图\其他文件\nb2\nb2\plugins\nonebot_plugin_lxns_maimai\render.py", line 13, in render_b50
    star_counts = await asyncio.gather(*tasks)
  File "D:\截图\其他文件\nb2\nb2\plugins\nonebot_plugin_lxns_maimai\utils.py", line 22, in calc_star_count
    song = await API.get_song_info(score.id)
  File "D:\截图\其他文件\nb2\nb2\plugins\nonebot_plugin_lxns_maimai\apis\request.py", line 74, in get_song_info
    return Song(**response.json())
  File "D:\截图\其他文件\nb2\.venv\Lib\site-packages\pydantic\main.py", line 253, in __init__
    validated_self = self.__pydantic_validator__.validate_python(data, self_instance=self)
pydantic_core._pydantic_core.ValidationError: 7 validation errors for Song
id
  Field required [type=missing, input_value={'success': False, 'code'...e': 'too many requests'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.11/v/missing
title
  Field required [type=missing, input_value={'success': False, 'code'...e': 'too many requests'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.11/v/missing
artist
  Field required [type=missing, input_value={'success': False, 'code'...e': 'too many requests'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.11/v/missing
genre
  Field required [type=missing, input_value={'success': False, 'code'...e': 'too many requests'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.11/v/missing
bpm
  Field required [type=missing, input_value={'success': False, 'code'...e': 'too many requests'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.11/v/missing
version
  Field required [type=missing, input_value={'success': False, 'code'...e': 'too many requests'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.11/v/missing
difficulties
  Field required [type=missing, input_value={'success': False, 'code'...e': 'too many requests'}, input_type=dict]
    For further information visit https://errors.pydantic.dev/2.11/v/missing
```

--- a/nonebot_plugin_lxns_maimai/apis/request.py
+++ b/nonebot_plugin_lxns_maimai/apis/request.py
@@ -74,6 +74,14 @@ class API:
         return Song(**response.json())
 
     @classmethod
+    async def get_song_list(cls, *, is_notes: bool = False) -> list[Song]:
+        async with httpx.AsyncClient() as client:
+            response = await client.get(
+                f"{url}/song/list" if not is_notes else f"{url}/song/list?notes=true"
+            )
+        return [Song(**song) for song in response.json()["songs"]]
+
+    @classmethod
     async def download_player_icon(cls, player: Player) -> BytesIO | bytes:
         if not player.icon:
             return BytesIO()

--- a/nonebot_plugin_lxns_maimai/render.py
+++ b/nonebot_plugin_lxns_maimai/render.py
@@ -1,5 +1,3 @@
-import asyncio
-
 from nonebot_plugin_htmlrender import template_to_pic
 
 from .schema import RenderProps
@@ -9,8 +7,7 @@ from .filters import star_count_to_color, level_index_to_color, course_rank_id_t
 
 
 async def render_b50(props: RenderProps) -> bytes:
-    tasks = [calc_star_count(score) for score in props.standard + props.dx]
-    star_counts = await asyncio.gather(*tasks)
+    star_counts = await calc_star_count(list(props.standard + props.dx))
 
     for score, star_count in zip(props.standard + props.dx, star_counts):
         score.star_count = star_count  # type: ignore

--- a/nonebot_plugin_lxns_maimai/utils.py
+++ b/nonebot_plugin_lxns_maimai/utils.py
@@ -16,31 +16,44 @@ def get_difficulty(
         return None
 
 
-async def calc_star_count(score: Score) -> int | None:
+async def calc_star_count(scores: list[Score]) -> list[int]:
     """计算 DX 分数星星数量"""
 
-    song = await API.get_song_info(score.id)
-    difficulty = get_difficulty(song, score.type, score.level_index)
+    song_list = await API.get_song_list(is_notes=True)
+    song_dict = {song.id: song for song in song_list}
 
-    if not difficulty or not difficulty.notes:
-        return None
+    matched_songs = []
+    for score in scores:
+        if song := song_dict.get(score.id):
+            matched_songs.append((song, score))
 
-    if isinstance(difficulty, SongDifficultyUtage):
-        percentage = (
-            score.dx_score / (difficulty.notes.left + difficulty.notes.right) * 3  # type: ignore
-        ) * 100
-    else:
-        percentage = score.dx_score / (difficulty.notes.total * 3) * 100  # type: ignore
+    star_counts = []
+    for song, score in matched_songs:
 
-    if percentage >= 97:
-        return 5
-    elif percentage >= 95:
-        return 4
-    elif percentage >= 93:
-        return 3
-    elif percentage >= 90:
-        return 2
-    elif percentage >= 85:
-        return 1
-    else:
-        return 0
+        difficulty = get_difficulty(song, score.type, score.level_index)
+
+        if not difficulty or not difficulty.notes:
+            star_counts.append(0)
+            continue
+
+        if isinstance(difficulty, SongDifficultyUtage):
+            percentage = (
+                score.dx_score / (difficulty.notes.left + difficulty.notes.right) * 3  # type: ignore
+            ) * 100
+        else:
+            percentage = score.dx_score / (difficulty.notes.total * 3) * 100  # type: ignore
+
+        if percentage >= 97:
+            star_counts.append(5)
+        elif percentage >= 95:
+            star_counts.append(4)
+        elif percentage >= 93:
+            star_counts.append(3)
+        elif percentage >= 90:
+            star_counts.append(2)
+        elif percentage >= 85:
+            star_counts.append(1)
+        else:
+            star_counts.append(0)
+
+    return star_counts

